documentation/user_docs/docs/Advanced_Multifit.rst

```diff
@@ -0,0 +1,74 @@
+
+The methods required for fitting objects with multifit are as follows:
+
+Fit functions
+*************
+
+The global function, and background function(s) if given, can be methods of the class or simply functions, with input argument form as described in detail in multifit help. The general format is:
+
+::
+
+        >> wcalc = my_function (w,p,c1,c2,...)
+
+If multifit is defined as a method of the class, then one can use the capability of nesting functions within the method to accept different fit function syntax. This is done, for example, for sqw objects when the fit functions to multifit, and equivalently multifit_func, are just a 1D/2D...4D Gaussian (according to the dimensionality of the sqw object).
+
+
+Utility methods
+---------------
+
+These are required to enable multifit to work with objects
+
+wout = mask (win, msk)
+~~~~~~~~~~~~~~~~~~~~~~
+
+A method that masks data points from further calculation. The output object must be a valid instance of the class in which the masked values have been removed in whatever sense the class requires.
+
+
+[y,var,msk] = sigvar_get(win)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A method that returns the intensity and variance arrays from the objects, along with a mask array that indicates which elements are to be retained (where elements of msk are true, the corresponding elements of y and var are retained). The output arrays y and var must have the same size and shape; msk must have the same number of elements (but can be a different shape). The array msk must be understood by the method 'mask' defined below.
+
+
+wsum = w1 + w2
+~~~~~~~~~~~~~~
+
+If a background function is provided, addition of objects must be defined as (requires overloading of the addition operator with a method named plus.m)
+
+
+
+**optional** x = sigvar_getx(win)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Get the corresponding x values to the y, var, msk arrays that are returned by sigvar_get.
+
+-if one dimensional i.e. single x coordinate per point:
+      x must be a single array, the same size as y and var
+-if n-dimensional i.e. n x-values per point:
+      x must be a cell array of arrays, one per x dimension, each the same size as y and var as returned by sigvar_get.
+
+This method replaces the need to have the method 'mask_points' described below, as  'sigvar_getx' will enable the masking function built in to multifit to be used. However, if mask_points exists, then it will have priority over the use of sigvar_getx.
+
+
+**optional** [msk, ok, mess] = mask_points (win, 'keep', xkeep, 'remove', xremove, 'mask', msk_in)
```

Comment:
Does optional mean that all mask values are assumed 1 if this method is not provided?